src/Forms/GridField/GridFieldEditButton.php

```diff
@@ -70,6 +70,8 @@ public function getUrl($gridField, $record, $columnName, $addState = true)
             'edit'
         );
 
+        $link = $gridField->addRequestToURL($link);
+
         if ($addState) {
             $link = $this->getStateManager()->addStateToURL($gridField, $link);
         }
```

Comment:
This isn't needed anymore if we're keeping the gridfield `addRequestToURL` method.
Remove this and deprecate the `$addState` argument.
```suggestion
```

Reply:
`addStateToURL` method just appends new state to URL, but it is not keeping previous requests.

Reply:
> This isn't needed anymore if we're keeping the gridfield `addRequestToURL` method. Remove this and deprecate the `$addState` argument.

When user sorts or filters a GridField we send POST request, so we don't have these elements in `getVars()`

Reply:
https://github.com/silverstripe/silverstripe-framework/pull/10331#discussion_r924039943
When user sorts or filters a GridField we send POST request, so we don't have these elements in `getVars()`